src/ui/Analytics/Analytics.ts

```diff
@@ -254,6 +257,22 @@ export class Analytics extends Component {
       let event = this.componentOptionsModel.getEventName(Model.eventTypes.changeOne + ComponentOptionsModel.attributesEnum.searchHub);
       this.bind.onRootElement(event, (args: IAttributeChangedEventArg) => this.handleSearchHubChanged(args));
     }
+
+    AnalyticsUtils.instance.setClientInformationProvider(this);
+  }
+
+  public get visitorId() {
+    return Cookie.get('visitorId');
+  }
+
+  public get clientId() {
```

Comment:
I'm not clear on why we need to write and expose these `public` getters. If someone wants access to these values, could they not read them from cookies? Piping them through the `Analytics` class seems unncessary?

Regarding creating the `clientId`, I think this would happen on initializing the `Analytics` class rather than needing to call this specific getter.

Reply:
The goal was to centralize analytics to `AnalyticsUtils` and implicitly make it impossible to read `clientId` from it when `Analytics` were missing by making `Analytics` implement `IClientInformationProvider`.

However, looking back, this is unnecessary since the cookie could be read directly from `AnalyticsUtils` in the first place.

I'll make `Analytics` create `clientId` when it's instantiated and I'll make `AnalyticsUtils` read it.